### ISSUE TITLE
I have met the same problem with #861

### ISSUE THREAD

**Issue by jzyzxx** [fyne-io/fyne#1599]
### Describe the bug:

#861 

### To Reproduce:
Steps to reproduce the behaviour:

### Example code:

```
cd D:\gopath\src\fyne.io\fyne\cmd\fyne_demo
go run main.go
```

### Device (please complete the following information):
 - OS: [windows10]
 - Version: [latest]
 - Go version: [e.g. 1.15.5]
 - Fyne version: [latest]

My go env:
```
set GO111MODULE=on
set GOARCH=amd64
set GOBIN=C:\Go\bin
set GOCACHE=C:\Users\DELL\AppData\Local\go-build
set GOENV=C:\Users\DELL\AppData\Roaming\go\env
set GOEXE=.exe
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOINSECURE=
set GOMODCACHE=D:\gopath\pkg\mod
set GONOPROXY=
set GONOSUMDB=
set GOOS=windows
set GOPATH=D:\gopath
set GOPRIVATE=
set GOPROXY=https://goproxy.cn,direct
set GOROOT=c:\go
set GOSUMDB=off
set GOTMPDIR=
set GOTOOLDIR=c:\go\pkg\tool\windows_amd64
set GCCGO=gccgo
set AR=ar
set CC=gcc
set CXX=g++
set CGO_ENABLED=1
set GOMOD=D:\gopath\src\fyne.io\fyne\go.mod
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0 -fdebug-prefix-map=C:\Users\DELL\AppData\Local\Temp\go-build288357630=/tmp/go-build -gno-record-gcc-switches
```
My gcc env:
```
Using built-in specs.
COLLECT_GCC=C:\TDM-GCC-64\bin\gcc.exe
COLLECT_LTO_WRAPPER=C:/TDM-GCC-64/bin/../libexec/gcc/x86_64-w64-mingw32/9.2.0/lto-wrapper.exe
Target: x86_64-w64-mingw32
Configured with: ../../../src/gcc-git-9.2.0/configure --build=x86_64-w64-mingw32 --enable-targets=all --enable-languages=ada,c,c++,fortran,lto,objc,obj-c++ --enable-libgomp --enable-lto --enable-graphite --enable-cxx-flags=-DWINPTHREAD_STATIC --disable-build-with-cxx --disable-build-poststage1-with-cxx --enable-libstdcxx-debug --enable-threads=posix --enable-version-specific-runtime-libs --enable-fully-dynamic-string --enable-libstdcxx-threads --enable-libstdcxx-time --with-gnu-ld --disable-werror --disable-nls --disable-win32-registry --enable-large-address-aware --disable-rpath --disable-symvers --prefix=/mingw64tdm --with-local-prefix=/mingw64tdm --with-pkgversion=tdm64-1 --with-bugurl=http://tdm-gcc.tdragon.net/bugs
Thread model: posix
gcc version 9.2.0 (tdm64-1)
```

**Comment by andydotxyz** [comment]
Are you sure that the TDB gcc is what is run in the path? It looks like it might not be...
Check what you are building for with:
```
$ which gcc
$ gcc -dumpmachine
```

**Comment by jzyzxx** [comment]
> Are you sure that the TDB gcc is what is run in the path? It looks like it might not be...
> Check what you are building for with:
> 
> ```
> $ which gcc
> $ gcc -dumpmachine
> ```

Thanks for your response.
```
which gcc:
which : 无法将“which”项识别为 cmdlet、函数、脚本文件或可运行程序的名称。请检查名称的拼写，如果包括路径，请确保路径正
确，然后再试一次。
所在位置 行:1 字符: 1
+ which gcc
+ ~~~~~
    + CategoryInfo          : ObjectNotFound: (which:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
gcc -dumpmachine:
x86_64-w64-mingw32
```

**Comment by jzyzxx** [comment]
My gcc help:
```
gcc --help
Usage: gcc.exe [options] file...
Options:
  -pass-exit-codes         Exit with highest error code from a phase.
  --help                   Display this information.
  --target-help            Display target specific command line options.
  --help={common|optimizers|params|target|warnings|[^]{joined|separate|undocumented}}[,...].
                           Display specific types of command line options.
  (Use '-v --help' to display command line options of sub-processes).
  --version                Display compiler version information.
  -dumpspecs               Display all of the built in spec strings.
  -dumpversion             Display the version of the compiler.
  -dumpmachine             Display the compiler's target processor.
  -print-search-dirs       Display the directories in the compiler's search path.
  -print-libgcc-file-name  Display the name of the compiler's companion library.
  -print-file-name=<lib>   Display the full path to library <lib>.
  -print-prog-name=<prog>  Display the full path to compiler component <prog>.
  -print-multiarch         Display the target's normalized GNU triplet, used as
                           a component in the library path.
  -print-multi-directory   Display the root directory for versions of libgcc.
  -print-multi-lib         Display the mapping between command line options and
                           multiple library search directories.
  -print-multi-os-directory Display the relative path to OS libraries.
  -print-sysroot           Display the target libraries directory.
  -print-sysroot-headers-suffix Display the sysroot suffix used to find headers.
  -Wa,<options>            Pass comma-separated <options> on to the assembler.
  -Wp,<options>            Pass comma-separated <options> on to the preprocessor.
  -Wl,<options>            Pass comma-separated <options> on to the linker.
  -Xassembler <arg>        Pass <arg> on to the assembler.
  -Xpreprocessor <arg>     Pass <arg> on to the preprocessor.
  -Xlinker <arg>           Pass <arg> on to the linker.
  -save-temps              Do not delete intermediate files.
  -save-temps=<arg>        Do not delete intermediate files.
  -no-canonical-prefixes   Do not canonicalize paths when building relative
                           prefixes to other gcc components.
  -pipe                    Use pipes rather than intermediate files.
  -time                    Time the execution of each subprocess.
  -specs=<file>            Override built-in specs with the contents of <file>.
  -std=<standard>          Assume that the input sources are for <standard>.
  --sysroot=<directory>    Use <directory> as the root directory for headers
                           and libraries.
  -B <directory>           Add <directory> to the compiler's search paths.
  -v                       Display the programs invoked by the compiler.
  -###                     Like -v but options quoted and commands not executed.
  -E                       Preprocess only; do not compile, assemble or link.
  -S                       Compile only; do not assemble or link.
  -c                       Compile and assemble, but do not link.
  -o <file>                Place the output into <file>.
  -pie                     Create a dynamically linked position independent
                           executable.
  -shared                  Create a shared library.
  -x <language>            Specify the language of the following input files.
                           Permissible languages include: c c++ assembler none
                           'none' means revert to the default behavior of
                           guessing the language based on the file's extension.

Options starting with -g, -f, -m, -O, -W, or --param are automatically
 passed on to the various sub-processes invoked by gcc.exe.  In order to pass
 other options on to these processes the -W<letter> options must be used.

For bug reporting instructions, please see:
<http://tdm-gcc.tdragon.net/bugs>.
```

**Comment by andydotxyz** [comment]
> $ which gcc

If you are using PowerShell then try `gcm gcc`.
Can you also please find out which `ld` is being used? `gcm ld`

**Comment by jzyzxx** [comment]
> gcm gcc
```
gcm gcc
CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     gcc.exe                                            0.0.0.0    C:\TDM-GCC-64\bin\gcc.exe

gcm ld
CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     ld.exe                                             0.0.0.0    C:\TDM-GCC-64\bin\ld.exe
```

**Comment by andydotxyz** [comment]
Can you please re-run the TDM setup wizard and check that you installed the recommended WingW64 libraries?

**Comment by jzyzxx** [comment]
> Can you please re-run the TDM setup wizard and check that you installed the recommended WingW64 libraries?

[http://49.233.3.151/static/images/shoot1.jpg](url)
[http://49.233.3.151/static/images/shoot2.jpg](url)

**Comment by jzyzxx** [comment]
> Can you please re-run the TDM setup wizard and check that you installed the recommended WingW64 libraries?

![shoot1](https://user-images.githubusercontent.com/24405119/100075450-e8b50f80-2e7a-11eb-9ca5-81156fc922a3.jpg)
![shoot2](https://user-images.githubusercontent.com/24405119/100075476-f1a5e100-2e7a-11eb-9ce5-8cdea9165387.jpg)

**Comment by andydotxyz** [comment]
Can you please include the latest version of the error in case there is any variation from the previous bug?

**Comment by jzyzxx** [comment]
```
PS D:\gopath\src\fyne.io\fyne\cmd\fyne_demo> go run main.go
# command-line-arguments
c:\go\pkg\tool\windows_amd64\link.exe: running gcc failed: exit status 1
C:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\DELL\AppData\Local\Temp\go-link-000624127\000004.o: in function `glfwSetCursor':
D:\gopath\pkg\mod\github.com\go-gl\glfw\v3.3\glfw@v0.0.0-20200625191551-73d3c3675aa3/glfw/src/input.c:820: undefined reference to `__assert_func'
C:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\DELL\AppData\Local\Temp\go-link-000624127\000004.o: in function `glfwSwapBuffers':
D:\gopath\pkg\mod\github.com\go-gl\glfw\v3.3\glfw@v0.0.0-20200625191551-73d3c3675aa3/glfw/src/context.c:646: undefined reference to `__assert_func'
C:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\DELL\AppData\Local\Temp\go-link-000624127\000004.o: in function `glfwExtensionSupported':
D:\gopath\pkg\mod\github.com\go-gl\glfw\v3.3\glfw@v0.0.0-20200625191551-73d3c3675aa3/glfw/src/context.c:680: undefined reference to `__assert_func'
C:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\DELL\AppData\Local\Temp\go-link-000624127\000004.o: in function `glfwGetProcAddress':
D:\gopath\pkg\mod\github.com\go-gl\glfw\v3.3\glfw@v0.0.0-20200625191551-73d3c3675aa3/glfw/src/context.c:746: undefined reference to `__assert_func'
C:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\DELL\AppData\Local\Temp\go-link-000624127\000004.o: in function `glfwGetInputMode':
D:\gopath\pkg\mod\github.com\go-gl\glfw\v3.3\glfw@v0.0.0-20200625191551-73d3c3675aa3/glfw/src/input.c:476: undefined reference to `__assert_func'
C:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\DELL\AppData\Local\Temp\go-link-000624127\000004.o:D:\gopath\pkg\mod\github.com\go-gl\glfw\v3.3\glfw@v0.0.0-20200625191551-73d3c3675aa3/glfw/src/input.c:501: more undefined references to `__assert_func' follow
collect2.exe: error: ld returned 1 exit status
```

**Comment by jzyzxx** [comment]
> Can you please include the latest version of the error in case there is any variation from the previous bug?

```
PS D:\gopath\src\fyne.io\fyne\cmd\fyne_demo> go run main.go
# command-line-arguments
c:\go\pkg\tool\windows_amd64\link.exe: running gcc failed: exit status 1
C:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\DELL\AppData\Local\Temp\go-link-000624127\000004.o: in function `glfwSetCursor':
D:\gopath\pkg\mod\github.com\go-gl\glfw\v3.3\glfw@v0.0.0-20200625191551-73d3c3675aa3/glfw/src/input.c:820: undefined reference to `__assert_func'
C:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\DELL\AppData\Local\Temp\go-link-000624127\000004.o: in function `glfwSwapBuffers':
D:\gopath\pkg\mod\github.com\go-gl\glfw\v3.3\glfw@v0.0.0-20200625191551-73d3c3675aa3/glfw/src/context.c:646: undefined reference to `__assert_func'
C:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\DELL\AppData\Local\Temp\go-link-000624127\000004.o: in function `glfwExtensionSupported':
D:\gopath\pkg\mod\github.com\go-gl\glfw\v3.3\glfw@v0.0.0-20200625191551-73d3c3675aa3/glfw/src/context.c:680: undefined reference to `__assert_func'
C:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\DELL\AppData\Local\Temp\go-link-000624127\000004.o: in function `glfwGetProcAddress':
D:\gopath\pkg\mod\github.com\go-gl\glfw\v3.3\glfw@v0.0.0-20200625191551-73d3c3675aa3/glfw/src/context.c:746: undefined reference to `__assert_func'
C:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\DELL\AppData\Local\Temp\go-link-000624127\000004.o: in function `glfwGetInputMode':
D:\gopath\pkg\mod\github.com\go-gl\glfw\v3.3\glfw@v0.0.0-20200625191551-73d3c3675aa3/glfw/src/input.c:476: undefined reference to `__assert_func'
C:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\DELL\AppData\Local\Temp\go-link-000624127\000004.o:D:\gopath\pkg\mod\github.com\go-gl\glfw\v3.3\glfw@v0.0.0-20200625191551-73d3c3675aa3/glfw/src/input.c:501: more undefined references to `__assert_func' follow
collect2.exe: error: ld returned 1 exit status
```

**Comment by andydotxyz** [comment]
This looks to be the issue reported in #26 - the resolution to that issue was to re-install the 64bit Go installation.
It seems like the C compilers may be working correctly but you could have the 32 bit Go installation by mistake.

**Comment by jzyzxx** [comment]
> This looks to be the issue reported in #26 - the resolution to that issue was to re-install the 64bit Go installation.
> It seems like the C compilers may be working correctly but you could have the 32 bit Go installation by mistake.
I have removed all packges and cache,then i run 'go mod tidy' and 'go run main.go' but nothing changed

**Comment by jzyzxx** [comment]
> This looks to be the issue reported in #26 - the resolution to that issue was to re-install the 64bit Go installation.
> It seems like the C compilers may be working correctly but you could have the 32 bit Go installation by mistake.

I have removed all packges and cache,then i run 'go mod tidy' and 'go run main.go' but nothing changed

**Comment by andydotxyz** [comment]
The suspected issue is Go itself, not it's cache. You'd need to uninstall Go and install making sure to use the 64-bit download.

**Comment by superbirdgz** [comment]
hi, I have the same issue, with MinGW, go version 1.15.5
PS C:\WINDOWS\system32> gcm ld

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     ld.exe                                             0.0.0.0    C:\MinGW\bin\ld.exe


PS C:\WINDOWS\system32> gcm gcc

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     gcc.exe                                            0.0.0.0    C:\MinGW\bin\gcc.exe

**Comment by jzyzxx** [comment]
> > This looks to be the issue reported in #26 - the resolution to that issue was to re-install the 64bit Go installation.
> > It seems like the C compilers may be working correctly but you could have the 32 bit Go installation by mistake.
> 
> I have removed all packges and cache,then i run 'go mod tidy' and 'go run main.go' but nothing changed

![version](https://user-images.githubusercontent.com/24405119/100170536-dc739580-2f00-11eb-96d9-45f03b4b5fb1.jpg)
I am quite sure my go arcture is amd64.

**Comment by jzyzxx** [comment]
> The suspected issue is Go itself, not it's cache. You'd need to uninstall Go and install making sure to use the 64-bit download.

T have re-installed the 64bit Go installation,but the problem was not solved

**Comment by superbirdgz** [comment]
@jzyzxx I tried reinstallation 1.15.5 but no luck.  then I tried 1.13.1 which is the same one installed in my macbook, it works now. am wondering if 1.15.5 is incompatible with the latest fyne.

**Comment by jzyzxx** [comment]
> @jzyzxx I tried reinstallation 1.15.5 but no luck. then I tried 1.13.1 which is the same one installed in my macbook, it works now. am wondering if 1.15.5 is incompatible with the latest fyne.

Thanks for your experence,i shound have try.

**Comment by jzyzxx** [comment]
> @jzyzxx I tried reinstallation 1.15.5 but no luck. then I tried 1.13.1 which is the same one installed in my macbook, it works now. am wondering if 1.15.5 is incompatible with the latest fyne.

Thanks for your experence,i should have a try.

**Comment by jzyzxx** [comment]
@andydotxyz @superbirdgz 
I can't contain my excitement that my problem has resolved when  i changed my go version from 1.15 to 1.13

**Comment by jzyzxx** [comment]
@andydotxyz @superbirdgz  Thank you again for your patient help
